### PR TITLE
feat: add support for rsk networks

### PR DESCRIPTION
**Pull request by grinry**
Adds support for RSK networks (mainnet and testnet)

### CODE REVIEW

**Comment by bh2smith** [comment]
I think it will be important to check if the Gnosis Safe provides support for these networks. At the moment, I am unaware of these networks being supported. Probably best to ask Gnosis Discord (under Safe Developer - https://discord.gg/tq6ygGha)

File: src/hooks/token.ts [hunk 1]
```diff
@@ -16,6 +16,15 @@ export const networkMap = new Map([
...
+  [30, "rsk"],
+  [31, "trsk"],
+]);
+
+const customNativeTokens = new Map([
+  [56, "BNB"],
+  [100, "xDai"],
+  [30, "RBTC"],
+  [31, "tRBTC"],
 ]);
```
Comment (bh2smith) [inline src/hooks/token.ts:28]: I like this! Might want to include all supported networks though. In order to keep all this in a single location I would suggest including these values directly in `networkMap` (for easier maintenance).

**Comment by grinry** [comment]
> I think it will be important to check if the Gnosis Safe provides support for these networks. At the moment, I am unaware of these networks being supported. Probably best to ask Gnosis Discord (under Safe Developer - https://discord.gg/tq6ygGha)

Officially they don't, but they have it under their radar. Smart-contracts are deployed and I'm preparing to run backend services and gnosis UI myself until Gnosis team will start supporting it officially :)

**Comment by bh2smith** [comment]
> > I think it will be important to check if the Gnosis Safe provides support for these networks. At the moment, I am unaware of these networks being supported. Probably best to ask Gnosis Discord (under Safe Developer - https://discord.gg/tq6ygGha)
> 
> Officially they don't, but they have it under their radar. Smart-contracts are deployed and I'm preparing to run backend services and gnosis UI myself until Gnosis team will start supporting it officially :)

That is good news. Do you happen to know an expected date for official support? I was also going to say, this change may make the app compatible, but there is another place where the different networks need to be set (of course this is for the safe services to be aware of them). I believe @schmanu might recall this other place where we need to set these values...

**Comment by schmanu** [comment]
> That is good news. Do you happen to know an expected date for official support? I was also going to say, this change may make the app compatible, but there is another place where the different networks need to be set (of course this is for the safe services to be aware of them). I believe @schmanu might recall this other place where we need to set these values...

I think if we want to be on the official listing on those networks we have to create an issue at [safe-apps-](https://github.com/gnosis/safe-apps-list)

But as these networks are not officially supported this app will have to be added via custom apps for now and it should work with these changes.

**Comment by grinry** [comment]
> That is good news. Do you happen to know an expected date for official support? I was also going to say, this change may make the app compatible, but there is another place where the different networks need to be set (of course this is for the safe services to be aware of them). I believe @schmanu might recall this other place where we need to set these values...

They weren't able to give me any ETA.
So there is enough time for me to fix merge conflicts and do some refactoring on PR, as I see that something changed when previous PR was merged 🙂

**Comment by bh2smith** [comment]
Yes we just did a big change adding NFT support - sorry about the conflicts. 

Have you checked if this works on your end? What do you think @schmanu - could we merge this if it doesn't affect the rest of the app?

**Comment by bh2smith** [comment]
I believe we have recently incorporated changes that support all networks without having to manually include them here. If this is not the case, feel free to reopen.

**Pull request closed without merging**